src/cas/pyodide-cas.ts

```diff
@@ -327,13 +390,26 @@ export function usePyodide() {
       // TODO: If the expression is only a single getter or something simple, don't call the CAS
       pythonExpression = `${expressionToPython(command.expression[1])}\n\t.subs({${substitutions}})\n\t.evalf()`
     } else if (command.expression[0] == 'Evaluate') {
-      if ((command.expression[2] + '').toLowerCase() == 'solve') {
+      // https://docs.sympy.org/latest/tutorial/simplification.html
+      let evaluation = (command.expression[2] + '').toLowerCase()
+      const evaluationParameters = evaluation.match(/\\left\((.*?)\\right/)
+      const evaluationArgument = evaluationParameters ? evaluationParameters[1] : ''
+
+      if (evaluation.includes('solve')) {
         let variablesToSolveFor: string[] = []
-        getterNames.forEach((getterName) => {
-          if (!command.gettersData.has(getterName)) {
-            variablesToSolveFor.push(getterName)
-          }
-        })
+        if (evaluation == 'solve') {
+          // Automatically Find Variables
+          getterNames.forEach((getterName) => {
+            if (!command.gettersData.has(getterName)) {
+              variablesToSolveFor.push(getterName)
+            }
+          })
+        } else {
+          // TODO: only 1 variable?
+          // variablesToSolveFor.push((command.expression[2] + '').split(',')[1])
+          variablesToSolveFor.push(evaluationArgument)
+        }
+
```

Comment:
Okay, this is awesome! Sure `evaluation.match(/\\left\((.*?)\\right/)` might be a bit of a hack, but on whole I'm happy to finally have a better 'solve' function. And then there are all those other neat functions that you implemented down there 👍

Reply:
Yea, it is a little bit of a hack, I would like it if the content inside of the custom EVALUATE `\\xrightarrow` you added were parsed as MathJSON. I am sure there is a way but I have not been able to do it. This will also remove the need for `KnownLatexFunctions`

Reply:
I took a look at this and there are two quite reasonable options:
1. Adding `\\solve`, `\\factor`, and `\\expand` to the dictionary like you suggested. If we do that, then we should also make it possible to call them like `\\solve(3x+5==5, x)`
2. Adding the ability to parse unknown functions and using that https://github.com/stefnotch/quantum-sheet/issues/33 However, for that we'll need to wait for the next compute engine version

Which option would you prefer? Or should we leave it for a future pull request?

Reply:
Okay, let's wait for another PR that way we can compare them side-by-side

Reply:
Alright, I opened an issue to track this so that we don't forget about it https://github.com/stefnotch/quantum-sheet/issues/34